cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -368,44 +354,35 @@ public Secret generateClientsPublicKeySecret() {
     }
 
     /**
-     * Generate the Secret containing CA self-signed certificate for internal communication.
-     * It also contains the private key-certificate (signed by internal CA) for each brokers for communicating
-     * internally with Zookeeper as well
+     * Generate the Secret containing just the self-signed CA certificate used
+     * for signing brokers certificates used for communication with clients
+     * It's useful for users to extract the certificate itself to put as trusted on the clients
      * @return The generated Secret
      */
-    public Secret generateBrokersInternalSecret() {
-        Base64.Encoder encoder = Base64.getEncoder();
-
+    public Secret generateClusterPublicKeySecret() {
         Map<String, String> data = new HashMap<>();
-        data.put("internal-ca.crt", encoder.encodeToString(internalCA.cert()));
-
-        for (int i = 0; i < replicas; i++) {
-            CertAndKey cert = internalCerts.get(KafkaCluster.kafkaPodName(cluster, i));
-            data.put(KafkaCluster.kafkaPodName(cluster, i) + ".key", encoder.encodeToString(cert.key()));
-            data.put(KafkaCluster.kafkaPodName(cluster, i) + ".crt", encoder.encodeToString(cert.cert()));
-        }
-        return createSecret(KafkaCluster.brokersInternalSecretName(cluster), data);
+        data.put("ca.crt", Base64.getEncoder().encodeToString(clusterCA.cert()));
+        return createSecret(KafkaCluster.clusterPublicKeyName(cluster), data);
```

Comment:
Exposing the cluster cert alone in this Secret is for allowing a user to get this cert, putting it into truststore on clients and validating server authentication when connecting to the cluster, right ? Do we want to call it "cluster-ca.crt" as we do in the Secret created in `generateBrokersSecret` ?

Reply:
Since this is expected to be used by the users I call it `ca.crt` as that is more standard within Kubernetes and OpenShift (check for example the token Secrets).